omod-common/src/main/java/org/openmrs/module/webservices/helper/ModuleFactoryWrapper.java

```diff
@@ -0,0 +1,159 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.module.webservices.helper;
+
+import org.openmrs.api.APIAuthenticationException;
+import org.openmrs.api.context.Context;
+import org.openmrs.module.Module;
+import org.openmrs.module.ModuleFactory;
+import org.openmrs.module.ModuleFileParser;
+import org.openmrs.module.ModuleUtil;
+import org.openmrs.module.web.WebModuleUtil;
+import org.openmrs.util.PrivilegeConstants;
+import org.springframework.web.multipart.MultipartFile;
+
+import javax.servlet.ServletContext;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.Collection;
+import java.util.List;
+
+/**
+ * ModuleFactoryWrapper wraps static methods of ModuleFactory to make code operating on modules
+ * testable
+ */
+public class ModuleFactoryWrapper {
```

Comment:
I'm okay we having this wrapper for methods that delegate multiple calls to other methods in core but I don't think you need methods on this wrapper that make a single call directly to the ModuleFactory or WebModuleUtil or ModuleUtil.